kalamine/template.py

```diff
@@ -387,18 +395,206 @@ def klc_deadkeys(layout: "KeyboardLayout") -> List[str]:
 
     output = []
     for k in DK_INDEX:
         if k not in layout.dead_keys:
             continue
         dk = layout.dead_keys[k]
-        output.append(f"{hex_ord(dk[' '])}\t\"{DK_INDEX[k].name.upper()}\"")
+        if format == "c":
+            output.append(
+                f"L\"\\\\x{hex_ord(dk[' '])}\"\tL\"{DK_INDEX[k].name.upper()}\","
+            )
+        else:
+            output.append(f"{hex_ord(dk[' '])}\t\"{DK_INDEX[k].name.upper()}\"")
     return output
 
 
+# This need the klc file to be created first
+def create_c_files(
+    layout: "KeyboardLayout",
+    msklc: Path,
+    verbose: bool,
+    working_dir: Path = Path(os.getcwd()),
+):
+    """Call kbdutools to generate C files"""
+    kbdutools = msklc / Path("bin/i386/kbdutool.exe")
+    cur = os.getcwd()
+    os.chdir(working_dir)
+    ret = subprocess.run(
+        [kbdutools, "-u", "-s", layout.meta["name8"] + ".klc"],
+        capture_output=not verbose,
+    )
+    os.chdir(cur)
+    ret.check_returncode()
+
+
+def c_keymap(layout: "KeyboardLayout") -> List[str]:
+    """Windows C layout, main part."""
+
+    global oem_idx
+    supported_symbols = "1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+    oem_idx = 0  # Python trick to do equivalent of C static variable
+    output = []
+    for key_name in LAYER_KEYS:
+        if key_name.startswith("-"):
+            continue
+
+        if key_name in ["ae13", "ab11"]:  # ABNT / JIS keys
+            continue  # these two keys are not supported yet
+
+        symbols = []
+        dead_symbols = []
+        alpha = False
+        has_dead = False
+
+        for i in [Layer.BASE, Layer.SHIFT, Layer.ALTGR, Layer.ALTGR_SHIFT]:
+            layer = layout.layers[i]
+
+            if key_name in layer:
+                symbol = layer[key_name]
+                desc = symbol
+                dead = "WCH_NONE"
+                if symbol in layout.dead_keys:
+                    desc = layout.dead_keys[symbol][" "]
+                    symbol = "WCH_DEAD"
+                    dead = hex_ord(desc)
+                    has_dead = True
+                else:
+                    if i == Layer.BASE:
+                        alpha = symbol.upper() != symbol
+                    if symbol not in supported_symbols:
+                        symbol = hex_ord(symbol)
+                symbols.append(symbol)
+                dead_symbols.append(dead)
+            else:
+                desc = " "
+                symbols.append("WCH_NONE")
+                dead_symbols.append("WCH_NONE")
+
+        virtual_key = klc_virtual_key(layout, symbols, "")
+        if len(virtual_key) == 1:
+            virtual_key = "'" + virtual_key + "'"
+        else:
+            virtual_key = "VK_" + virtual_key
+
+        def process_symbol(symbol: str) -> str:
+            if len(symbol) == 4:
+                return "0x" + symbol
+            elif len(symbol) == 1:
+                return "'" + symbol + "'"
```

Comment:
nit: `if` instead of `elif` (the linter should catch that…) 
and please use f-strings while we’re at it :-)